### PR TITLE
feat(topology): additional metrics for applying change operations

### PR DESCRIPTION
**Pull request by lenaschoenburg**
Adds counters for failed and applied operations as well as a gauge that times the time it takes to (successfully or not) apply an operation.

Relates to #15172

### CODE REVIEW

File: topology/src/main/java/io/camunda/zeebe/topology/metrics/TopologyMetrics.java [hunk 1]
```diff
@@ -55,6 +61,20 @@ public final class TopologyMetrics {
...
+          .namespace(NAMESPACE)
+          .name("cluster_changes_operation_duration")
+          .help("Duration it takes to apply an operation")
+          .labelNames(LABEL_OPERATION)
+          .register();
+  private static final Counter OPERATION_ATTEMPTS =
+      Counter.build()
+          .namespace(NAMESPACE)
+          .name("cluster_changes_operation_attempts")
+          .help("Number of retries per operation type")
```
Comment (deepthidevaki) [inline topology/src/main/java/io/camunda/zeebe/topology/metrics/TopologyMetrics.java:75]: :thinking: This not number of retries, right? It is more like "Number of attempts (failed or applied) per operation type"

File: topology/src/main/java/io/camunda/zeebe/topology/metrics/TopologyMetrics.java [hunk 2]
```diff
@@ -55,6 +61,20 @@ public final class TopologyMetrics {
           .name("cluster_changes_operations_completed")
           .help("Number of completed changes in the current change plan")
           .register();
+  private static final Gauge OPERATION_DURATION =
+      Gauge.build()
```
Comment (deepthidevaki) [inline topology/src/main/java/io/camunda/zeebe/topology/metrics/TopologyMetrics.java:65]: :question: Should this be a histogram? How do you plan to visualize this?

**Review by deepthidevaki — APPROVED** [review]
:rocket: Can't wait to see it all visualized in Grafana.

File: topology/src/main/java/io/camunda/zeebe/topology/metrics/TopologyMetrics.java (hunk 2 above)
Comment (oleschoenburg) [reply topology/src/main/java/io/camunda/zeebe/topology/metrics/TopologyMetrics.java]: I thought a histogram was too much but I'm not sure how to visualize with a gauge. For backups we also used a histogram so I changed it now :+1:

**Comment by oleschoenburg** [comment]
bors merge

File: topology/src/main/java/io/camunda/zeebe/topology/metrics/TopologyMetrics.java (hunk 2 above)
Comment (deepthidevaki) [reply topology/src/main/java/io/camunda/zeebe/topology/metrics/TopologyMetrics.java]: We can revisit when we add the visualization.

**Pull request merged**